### ISSUE TITLE
Topic name not included in messages for forum chats

### ISSUE THREAD

**Issue by alexvirlan** [tdlib/telegram-bot-api#356]
**When receiving a message from a forum type of chat, the topic name is not included.** Only the 'MessageThreadId' int is available that can be used to send a message in that specific topic. The topic name comes only when an event like 'ForumTopicCreated' is fired.

I think that this (**also receiving the topic name along with the 'MessageThreadId'**) would be a great feature to have for most, if not all, bots that are used in forums. I have a few bots on my own and some feedback from some friends that feel the same.
Please take this into consideration! Thanks!

Additional info: If a chat has the 'IsForum' property to true, the default/general topic will not set the 'IsTopicMessage' bool to true. Only if the message comes from any other topic. And by using the 'MessageThreadId' I can send messages to different topics but I don't have a way of knowing their name.

**Comment by levlam** [comment]
> If a chat has the 'IsForum' property to true, the default/general topic will not set the 'IsTopicMessage' bool to true.

This is correct. All messages not belonging to a topic are shown by apps in the "General" topic.

> Only if the message comes from any other topic. And by using the 'MessageThreadId' I can send messages to different topics but I don't have a way of knowing their name.

In supergroups messages can be split by threads. Each message thread has a unique identifier and consists of a root message, which isn't a reply to another message, and all messages, which are directly or indirectly reply to the root message. If a message thread is in a forum supergroup and its root message is a service message about forum topic creation, then the message thread forms a forum topic. If a message thread is in a discussion supergroup and its root message is an automatically forwarded post from the linked channel, then the message thread contains comments to the post. Otherwise, it is just a generic message thread for now. `message.message_thread_id` contains identifier of the message thread to which belongs the message, and this thread is a forum topic only if `message.is_topic_message == true`. Messages from "General" topic can form their own generic message threads, which are not forum topics.

**Comment by alexvirlan** [comment]
Thank you @levlam for taking the time to explain how it works!

**Comment by battleoftwok** [comment]
> > If a chat has the 'IsForum' property to true, the default/general topic will not set the 'IsTopicMessage' bool to true.
> 
> This is correct. All messages not belonging to a topic are shown by apps in the "General" topic.
> 
> > Only if the message comes from any other topic. And by using the 'MessageThreadId' I can send messages to different topics but I don't have a way of knowing their name.
> 
> In supergroups messages can be split by threads. Each message thread has a unique identifier and consists of a root message, which isn't a reply to another message, and all messages, which are directly or indirectly reply to the root message. If a message thread is in a forum supergroup and its root message is a service message about forum topic creation, then the message thread forms a forum topic. If a message thread is in a discussion supergroup and its root message is an automatically forwarded post from the linked channel, then the message thread contains comments to the post. Otherwise, it is just a generic message thread for now. `message.message_thread_id` contains identifier of the message thread to which belongs the message, and this thread is a forum topic only if `message.is_topic_message == true`. Messages from "General" topic can form their own generic message threads, which are not forum topics.

### Why, then, is the information provided on the official website different?
https://core.telegram.org/api/forum :
> To fetch the topic list of a forum, use [channels.getForumTopics](https://core.telegram.org/method/channels.getForumTopics); the same method can be used to search topics by their name.

**Comment by levlam** [comment]
@battleoftwok 
The on-site documentation is correct. It explain that "General" topic isn't a message thread and must not be used as such. By the way, you are reading documentation for app developers, which isn't related to bots.

**Comment by battleoftwok** [comment]
> @battleoftwok The on-site documentation is correct. It explain that "General" topic isn't a message thread and must not be used as such. By the way, you are reading documentation for app developers, which isn't related to bots.

So I really can't get the forum topic name? Just because the forum is a single chat (thread), and the topics in this forum are parts of this single chat? When creating a topic, the user enters a topic name, but it's not saved?

**Comment by levlam** [comment]
Bots can't get topic name through Bot API. Currently, they can only save topic name from "forum_topic_created" and "forum_topic_edited" messages as the closest approximation.

**Comment by battleoftwok** [comment]
> Bots can't get topic name through Bot API. Currently, they can only save topic name from "forum_topic_created" and "forum_topic_edited" messages as the closest approximation.

Obviously, the only way to get around this crutch and get the name of the newly created topic is to write your own: allow the user to send a request to the bot to create a new topic, passing it the name and other necessary data. Thank you!

**Comment by cha0sg0d** [comment]
For anyone interested, I discovered that you can log the topic name via `ctx.message.forum_topic_created?.name`;

**Comment by battleoftwok** [comment]
> For anyone interested, I discovered that you can log the topic name via `ctx.message.reply_to_message?.forum_topic_created?.name`;

The option you suggested does not suit my requirements. You can find out the name of the topic during its creation. However, I already have a form that contains many topics created before the bot was added there. Therefore, the bot will no longer receive their names.

**Comment by RememberTheAir** [comment]
Actually @cha0sg0d's suggestion can be helpful for your use case. Every message sent in a topic (that is not the "general" topic) is shipped with the `reply_to_message` property. This property will either contain the replied-to message (if the user that sent the message replied to another message in that topic), or the service message about the topic's creation (because messages sent in a topic that are not a reply to another message are shipped like that, as a reply to a "topic created" service message). In this case, you can use the `Message.reply_to_message.forum_topic_created.name` property to locally update your topic names' database

**Comment by duytbp** [comment]
`Message.reply_to_message.forum_topic_created.name` is the original topic name, which could be later changed by members

**Comment by johnsmith3321** [comment]
Do you want to download Telegram Premium without any cost?

Then visit here - [https://blackapks.com/en/telegram-premium](https://blackapks.com/en/telegram-premium)
